cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -201,12 +203,10 @@ func (cm *certManagerClient) install(ctx context.Context, version string, objs [
 // a cert-manager upgrade if necessary.
 func (cm *certManagerClient) PlanUpgrade(ctx context.Context) (CertManagerUpgradePlan, error) {
 	log := logf.Log
-
-	objs, err := cm.proxy.ListResources(ctx, map[string]string{clusterctlv1.ClusterctlCoreLabel: clusterctlv1.ClusterctlCoreLabelCertManagerValue}, certManagerNamespace)
+	objs, err := cm.getFilteredCertManagerResources(ctx)
 	if err != nil {
-		return CertManagerUpgradePlan{}, errors.Wrap(err, "failed get cert manager components")
+		return CertManagerUpgradePlan{}, err
```

Comment:
I think we want to wrap this with some context, e.g., "failed to calculate relevant cert-manager resources."

Reply:
the error is wrapped already by `getFilteredCertManagerResources` do you think changing the wrap text there would be helpful?

Reply:
Ok, I see now.

Given that the context we're adding `"failed to calculate cert-manager components for upgrade"`,  I think we it makes more sense to add it here, at the call site, rather than in the `getFilteredCertManagerResources` function. That's because the `proxy.ListResources` call doesn't calculate cert-manager components, but lists them. If we want to, we wrap that error with `"failed to list resources"`.

Reply:
makes sense to me. i'll update the pr.

Reply:
Thank you! :pray: